### ISSUE TITLE
why need a focalloss or CrossEntropyLoss after amsoftmax loss head?

### ISSUE THREAD

**Issue by mlinxiang** [ZhaoJ9014/face.evoLVe.PyTorch#46]

**Comment by ZhaoJ9014** [comment]
> _No description provided._

Hi, thank you for your question. Focal loss focus on the hard samples which benefit the learning process and overall overall performance with arcface.

**Comment by mlinxiang** [comment]
i am confused by why need two loss, do the first softmax loss act as a normlization?

**Comment by ShiinaMitsuki** [comment]
@mlinxiang 
原始loss函数
![image](https://user-images.githubusercontent.com/16696202/69928346-4d52be00-14f6-11ea-95b1-a517ac6d56bd.png)
改进的loss函数
![image](https://user-images.githubusercontent.com/16696202/69928363-5f346100-14f6-11ea-9bb1-85ff753cb98a.png)
如你所见，并不是两个loss函数，只是改变了每一项的计算方式。
